### DISCUSSION TITLE
How can I get the file count in size linemode?

### DISCUSSION THREAD

**Discussion by TobisLee** [sxyazi/yazi#1563] [1. Q&A] [answered]
I tried to solve #1277 and found the following code used to calculate the folder size:
https://github.com/sxyazi/yazi/blob/caacd1511085a3da1ecef61a73df98c9e5268238/yazi-fm/src/lives/file.rs#L41-L43

I added the following code to get the file count in a folder:
```rust
reg.add_method("len_or_size", |_, me, ()| {
	Ok(if me.is_dir() { Some(me.folder().files.len() as u64) } else { Some(me.len) })
});
```

However, `me.folder()` returns the parent folder of `me`, not `me` itself. I noticed that `me` is a `File` struct, I couldn't find a method or property to get the file count within the `File` struct. Is there any way to get the file count in this method?

**Comment by AnirudhG07** [comment]
Currently, no such method exists, as per [docs](https://yazi-rs.github.io/docs/plugins/utils). You could make plugin which gives you the file_count.

**Comment by TobisLee** [comment, marked as answer]
For now, my workaround is to use `fs::read_dir` to calculate the file count. See details at: https://github.com/TobisLee/yazi/commit/aa965199f5c2c5ec03d393db5e9fc83bc41de0e8

**Reply by TobisLee** [reply]
Hey, @AnirudhG07, I don't find your reply, I just put the screenshot here.
![图片](https://github.com/user-attachments/assets/8abd3f19-c8d5-469a-b63e-89129d7b959d)

**Comment by TobisLee** [comment]


**Reply by AnirudhG07** [reply]
No I deleted it. Cause you showed an example in 1 of the PR's. Still thanks!

**Reply by TobisLee** [reply]
All right. No problem.
